src/zones/dbm.rs

```diff
@@ -676,6 +676,79 @@ impl DBM<Valid> {
         unsafe { dbm.assert_valid() }
     }
 
+    fn compute_tables(&self, src_clocks: &Vec<bool>, dst_clocks: &Vec<bool>) -> (Vec<ClockIndex>, Vec<ClockIndex>) {
+        let mut dst_to_src = Vec::<ClockIndex>::default();
+        let mut src_to_dst = Vec::<ClockIndex>::default();
+        let mut src_ind = 0;
+        assert_eq!(src_clocks.len(), dst_clocks.len());
+
+        for (src, dst) in src_clocks.iter().zip(dst_clocks.iter()){
+            let (src, dst) = (src.to_owned(), dst.to_owned());
+            if dst {
+                src_to_dst.push(dst_to_src.len());
+                if src {
+                    dst_to_src.push(src_ind);
+                }
+                else {
+                    dst_to_src.push(0);
+                }
+            }
+            else {
+                src_to_dst.push(0);
+            }
+
+            src_ind += 1;
+        };
+
+        return (src_to_dst, dst_to_src);
+    }
+
+    fn update_dbm(&mut self, src: &DBM<Valid>, dst_to_src: &Vec<ClockIndex>) {
+        self.data[0] = LE_ZERO;
+
+        for j in 1..self.dim {
+            if dst_to_src[j] == 0
+            {
+                self.data[j] = LE_ZERO;
+            }
+            else
+            {
+                self.data[j] = src.data[dst_to_src[j]];
+            }
+        }
+
+        for i in 1..self.dim {
+            if dst_to_src[i] != 0 { // If copy from src.
+                let constraint0 = src.data[src.dim * dst_to_src[i]];
+                self.data[i * self.dim] = constraint0;
+
+                for j in 1..self.dim {
+                    if dst_to_src[j] == 0
+                    {
+                        self.data[j + i * self.dim] = constraint0;

```

Comment:
Thinks this is the cause for the numbers that arent infinity when you expand

Reply:
Yes, i am aware of that. I dont know why UDBM decided to do it like this though. Thats what i was talking about at the meeting yesterday.

Reply:
Not to make it an entire topic, but in the process of expanding, when the new row is created the for loop still iterates over the other axis, i suspect that it carries this line over to the expanded dimension and overwrites the infinity. Just a thought, but you have my stamp of approval

Reply:
No, this line only gets run if the clock is in the src. There should be an if statement pretty much directly above the for loop in your comment.